docs/BUILD.md

```diff
@@ -1,6 +1,36 @@
+
 # Building s2n-tls
 
-s2n-tls can be built as follows:
+To use s2n-tls, you must build the library from the source and then include it in your program.
+
+## Requirements
+
+s2n-tls supports and tests on **x86** and **arm** architectures.
+
+### System requirements
+
+* 20GB RAM available, recommended.
+
+### Software requirements
+
+Building s2n-tls requires:
+
+1. Git
+1. GCC or Clang
+1. CMake
+1. Libcrypto, such as OpenSSL
+1. Platform-specific build tools
+
+## Building s2n-tls from the source
+
+Clone s2n-tls and change directories into the repo:
+
+```bash
+git clone https://github.com/aws/s2n-tls.git
+cd s2n-tls
```

Comment:
I think we wanted to avoid putting git clone instructions everywhere since they're very basic and also not really related to building.

https://github.com/aws/s2n-tls/pull/4126#discussion_r1282376301

Reply:
Yes, it is basic. It's also a common docs convention. Think of it as starting the story at the very beginning.